discovery_engine/test/integration/feed_test.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io' show Directory;
+
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show
+        ClientEventSucceeded,
+        DiscoveryEngine,
+        NextFeedBatchRequestSucceeded,
+        RestoreFeedSucceeded;
+
+import '../logging.dart' show setupLogging;
+import 'utils/helpers.dart'
+    show TestEngineData, expectEvent, initEngine, setupTestEngineData;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  group('DiscoveryEngine feed', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    late DiscoveryEngine engine;
+
+    setUp(() async {
+      data = await setupTestEngineData();
+      server = await LocalNewsApiServer.start();
+      engine = await initEngine(data, server.port);
+    });
+
+    tearDown(() async {
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('close documents', () async {
+      final batch = expectEvent<NextFeedBatchRequestSucceeded>(
+        await engine.requestNextFeedBatch(),
+      ).items;
+      //FIXME once we have mock data producing multiple batches close muiltiple but not all documents
```

Comment:
:pushpin: We should extend the mock data to have multiple batches and more then just one document per batch. But that is outside of scope.

Reply:
:art: for clarity, can you extend this `FIXME` comment to also later uncomment the other assertions below? just to make it clear those assertions relate to this fix